### DISCUSSION TITLE
Custom pattern and better `react-hook-form` integration for `PinInput`

### DISCUSSION THREAD

**Discussion by AlphaNecron** [chakra-ui/chakra-ui#9076] [General]
### Description

1. Current `PinInput` doesn't accept custom validation rules. I was able to alter the default behavior by copying the entire `PinInput` component and customize it, which seems way too complicated or even impossible at the moment due to reference to `ark` -> `zag`.
2. `react-hook-form` doesn't work well with `PinInput` (when using `string` as value type), the only way I think of is to passing field props to `HiddenInput` but the value is not reflected when `react-hook-form` is updated.

### Problem Statement/Justification

1. Not everyone wants to stick with statically predefined patterns.
2. `PinInput` should be able to work with `string` value according to its use case tho, just like `v2`.

### Proposed Solution or API

1. Allow setting `type` to something like `none` and let users customize the behavior using `pattern` and/or `onValueChange`.
2. Underlying string array should be updated when `HiddenInput`'s `onChange` is invoked.

**Comment by AlphaNecron** [comment]
Bump

**Comment by ndchinh2706** [comment]
Bump

**Comment by segunadebayo** [comment]
I've added an example to the docs for this:
https://www.chakra-ui.com/docs/components/pin-input#hook-form

**Reply by AlphaNecron** [reply]
Thanks! How about custom pattern thingy?

**Reply by AlphaNecron** [reply]
Bump, overriding default pattern is still unimplemented.

**Comment by segunadebayo** [comment]
Closing as stale. Please start a new discussion if this is still relevant on the latest version.
